### PR TITLE
Fix/workflow

### PR DESCRIPTION
**Pull request by TerminalFi**
Fixing various issues with the previous workflow. 

Also tested locally with the tool `act`
<details>
  <summary>Local Test Output</summary>
  
```bash
0DOCKER_HOST=unix:///Users/$(whoami)/Library/Containers/com.docker.docker/Data/docker.raw.sock act
1[Gopls Update Checker/compareTag] 🚀  Start image=catthehacker/ubuntu:act-latest
2[Gopls Update Checker/compareTag]   🐳  docker pull image=catthehacker/ubuntu:act-latest platform= username= forcePull=false
3[Gopls Update Checker/compareTag]   🐳  docker create image=catthehacker/ubuntu:act-latest platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
4[Gopls Update Checker/compareTag]   🐳  docker run image=catthehacker/ubuntu:act-latest platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
5[Gopls Update Checker/compareTag] ⭐ Run Main Checkout code
6[Gopls Update Checker/compareTag]   🐳  docker cp src=/Users/TerminalFi/Library/Application Support/Sublime Text/Packages/LSP-gopls/. dst=/Users/TerminalFi/Library/Application Support/Sublime Text/Packages/LSP-gopls
7[Gopls Update Checker/compareTag]   ✅  Success - Main Checkout code
8[Gopls Update Checker/compareTag] ⭐ Run Main Run script
9[Gopls Update Checker/compareTag]   🐳  docker exec cmd=[python /var/run/act/workflow/tagCompare] user= workdir=
10[Gopls Update Checker/compareTag]   ✅  Success - Main Run script
11[Gopls Update Checker/compareTag]   ⚙  ::set-output:: REQUIRES_UPDATE=0
12[Gopls Update Checker/compareTag]   ⚙  ::set-output:: LATEST_VERSION=0.11.0
13[Gopls Update Checker/compareTag]   ⚙  ::set-output:: BRANCH_NAME=0_11_0
14[Gopls Update Checker/compareTag] 🏁  Job succeeded
15[Gopls Update Checker/updateVersion] 🚀  Start image=catthehacker/ubuntu:act-latest
16[Gopls Update Checker/updateVersion]   🐳  docker pull image=catthehacker/ubuntu:act-latest platform= username= forcePull=false
17[Gopls Update Checker/updateVersion]   🐳  docker create image=catthehacker/ubuntu:act-latest platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
18[Gopls Update Checker/updateVersion]   🐳  docker run image=catthehacker/ubuntu:act-latest platform= entrypoint=["tail" "-f" "/dev/null"] cmd=[]
19[Gopls Update Checker/updateVersion]   ☁  git clone 'https://github.com/actions/setup-go' # ref=v3
20[Gopls Update Checker/updateVersion]   ☁  git clone 'https://github.com/actions/setup-python' # ref=v4
21[Gopls Update Checker/updateVersion] ⭐ Run Main actions/checkout@v3
22[Gopls Update Checker/updateVersion]   🐳  docker cp src=/Users/TerminalFi/Library/Application Support/Sublime Text/Packages/LSP-gopls/. dst=/Users/TerminalFi/Library/Application Support/Sublime Text/Packages/LSP-gopls
23[Gopls Update Checker/updateVersion]   ✅  Success - Main actions/checkout@v3
24[Gopls Update Checker/updateVersion] ⭐ Run Main actions/setup-go@v3
25[Gopls Update Checker/updateVersion]   🐳  docker cp src=/Users/TerminalFi/.cache/act/actions-setup-go@v3/ dst=/var/run/act/actions/actions-setup-go@v3/
26[Gopls Update Checker/updateVersion]   🐳  docker exec cmd=[node /var/run/act/actions/actions-setup-go@v3/dist/setup/index.js] user= workdir=
27| Setup go version spec 1.19
28| Attempting to resolve the latest version from the manifest...
29[Gopls Update Checker/updateVersion]   💬  ::debug::No manifest cached
30| matching 1.19...
31[Gopls Update Checker/updateVersion]   💬  ::debug::check 1.19.4 satisfies 1.19
32[Gopls Update Checker/updateVersion]   💬  ::debug::x64===x64 && darwin===linux
33[Gopls Update Checker/updateVersion]   💬  ::debug::x64===x64 && linux===linux
34[Gopls Update Checker/updateVersion]   💬  ::debug::matched 1.19.4
35| Resolved as '1.19.4'
36[Gopls Update Checker/updateVersion]   💬  ::debug::isExplicit: 1.19.4
37[Gopls Update Checker/updateVersion]   💬  ::debug::explicit? true
38[Gopls Update Checker/updateVersion]   💬  ::debug::checking cache: /opt/hostedtoolcache/go/1.19.4/x64
39[Gopls Update Checker/updateVersion]   💬  ::debug::not found
40| Attempting to download 1.19.4...
41[Gopls Update Checker/updateVersion]   💬  ::debug::No manifest cached
42| matching 1.19.4...
43[Gopls Update Checker/updateVersion]   💬  ::debug::check 1.19.4 satisfies 1.19.4
44[Gopls Update Checker/updateVersion]   💬  ::debug::x64===x64 && darwin===linux
45[Gopls Update Checker/updateVersion]   💬  ::debug::x64===x64 && linux===linux
46[Gopls Update Checker/updateVersion]   💬  ::debug::matched 1.19.4
47| Acquiring 1.19.4 from https://github.com/actions/go-versions/releases/download/1.19.4-3636993500/go-1.19.4-linux-x64.tar.gz
48[Gopls Update Checker/updateVersion]   💬  ::debug::Downloading https://github.com/actions/go-versions/releases/download/1.19.4-3636993500/go-1.19.4-linux-x64.tar.gz
49[Gopls Update Checker/updateVersion]   💬  ::debug::Destination /tmp/5b423f3c-db5c-4b8b-b1e7-58980dd9d0a6
50[Gopls Update Checker/updateVersion]   💬  ::debug::download complete
51| Extracting Go...
52[Gopls Update Checker/updateVersion]   💬  ::debug::Checking tar --version
53[Gopls Update Checker/updateVersion]   💬  ::debug::tar (GNU tar) 1.34%0ACopyright (C) 2021 Free Software Foundation, Inc.%0ALicense GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.%0AThis is free software: you are free to change and redistribute it.%0AThere is NO WARRANTY, to the extent permitted by law.%0A%0AWritten by John Gilmore and Jay Fenlason.
54| [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /tmp/bbbd2f49-93d0-4399-ba5a-0176f190fe3f -f /tmp/5b423f3c-db5c-4b8b-b1e7-58980dd9d0a6
55| Successfully extracted go to /tmp/bbbd2f49-93d0-4399-ba5a-0176f190fe3f
56| Adding to the cache ...
57[Gopls Update Checker/updateVersion]   💬  ::debug::Caching tool go 1.19.4 x64
58[Gopls Update Checker/updateVersion]   💬  ::debug::source dir: /tmp/bbbd2f49-93d0-4399-ba5a-0176f190fe3f
59[Gopls Update Checker/updateVersion]   💬  ::debug::destination /opt/hostedtoolcache/go/1.19.4/x64
60[Gopls Update Checker/updateVersion]   💬  ::debug::finished caching tool
61| Successfully cached go to /opt/hostedtoolcache/go/1.19.4/x64
62| Added go to the path
63[Gopls Update Checker/updateVersion]   💬  ::debug::which go :/opt/hostedtoolcache/go/1.19.4/x64/bin/go:
64[Gopls Update Checker/updateVersion]   💬  ::debug::go env GOPATH :/root/go:
65[Gopls Update Checker/updateVersion]   💬  ::debug::creating /root/go
66[Gopls Update Checker/updateVersion]   💬  ::debug::creating /root/go/bin
67[Gopls Update Checker/updateVersion]   💬  ::debug::add bin true
68| Successfully set up Go version 1.19
69[Gopls Update Checker/updateVersion]   🚧  ::warning::The runner was not able to contact the cache service. Caching will be skipped
70[Gopls Update Checker/updateVersion]   ❓  ##[add-matcher]/run/act/actions/actions-setup-go@v3/matchers.json
71| go version go1.19.4 linux/amd64
72|
73[Gopls Update Checker/updateVersion]   ❓  ::group::go env
74| GO111MODULE=""
75| GOARCH="amd64"
76| GOBIN=""
77| GOCACHE="/root/.cache/go-build"
78| GOENV="/root/.config/go/env"
79| GOEXE=""
80| GOEXPERIMENT=""
81| GOFLAGS=""
82| GOHOSTARCH="amd64"
83| GOHOSTOS="linux"
84| GOINSECURE=""
85| GOMODCACHE="/root/go/pkg/mod"
86| GONOPROXY=""
87| GONOSUMDB=""
88| GOOS="linux"
89| GOPATH="/root/go"
90| GOPRIVATE=""
91| GOPROXY="https://proxy.golang.org,direct"
92| GOROOT="/opt/hostedtoolcache/go/1.19.4/x64"
93| GOSUMDB="sum.golang.org"
94| GOTMPDIR=""
95| GOTOOLDIR="/opt/hostedtoolcache/go/1.19.4/x64/pkg/tool/linux_amd64"
96| GOVCS=""
97| GOVERSION="go1.19.4"
98| GCCGO="gccgo"
99| GOAMD64="v1"
100| AR="ar"
101| CC="gcc"
102| CXX="g++"
103| CGO_ENABLED="1"
104| GOMOD="/dev/null"
105| GOWORK=""
106| CGO_CFLAGS="-g -O2"
107| CGO_CPPFLAGS=""
108| CGO_CXXFLAGS="-g -O2"
109| CGO_FFLAGS="-g -O2"
110| CGO_LDFLAGS="-g -O2"
111| PKG_CONFIG="pkg-config"
112| GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build1481341060=/tmp/go-build -gno-record-gcc-switches"
113|
114[Gopls Update Checker/updateVersion]   ❓  ::endgroup::
115[Gopls Update Checker/updateVersion]   ✅  Success - Main actions/setup-go@v3
116[Gopls Update Checker/updateVersion]   ⚙  ::set-output:: go-version=1.19.4
117[Gopls Update Checker/updateVersion]   ⚙  ::add-path:: /opt/hostedtoolcache/go/1.19.4/x64/bin
118[Gopls Update Checker/updateVersion]   ⚙  ::add-path:: /root/go/bin
119[Gopls Update Checker/updateVersion] ⭐ Run Main Install GOPLS
120[Gopls Update Checker/updateVersion]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/2] user= workdir=
121| go: downloading golang.org/x/tools/gopls v0.11.0
122| go: downloading golang.org/x/tools v0.5.0
123| go: downloading golang.org/x/tools v0.3.1-0.20221213193459-ca17b2c27ca8
124| go: downloading github.com/sergi/go-diff v1.1.0
125| go: downloading honnef.co/go/tools v0.3.3
126| go: downloading mvdan.cc/gofumpt v0.4.0
127| go: downloading mvdan.cc/xurls/v2 v2.4.0
128| go: downloading golang.org/x/mod v0.7.0
129| go: downloading golang.org/x/sys v0.2.0
130| go: downloading golang.org/x/sync v0.1.0
131| go: downloading golang.org/x/text v0.4.0
132| go: downloading golang.org/x/vuln v0.0.0-20221109205719-3af8368ee4fe
133| go: downloading github.com/google/go-cmp v0.5.9
134| go: downloading golang.org/x/exp/typeparams v0.0.0-20221031165847-c99f073a8326
135| go: downloading golang.org/x/exp v0.0.0-20221031165847-c99f073a8326
136| go: downloading github.com/BurntSushi/toml v1.2.1
137[Gopls Update Checker/updateVersion]   ✅  Success - Main Install GOPLS
138[Gopls Update Checker/updateVersion] ⭐ Run Main actions/setup-python@v4
139[Gopls Update Checker/updateVersion]   🐳  docker cp src=/Users/TerminalFi/.cache/act/actions-setup-python@v4/ dst=/var/run/act/actions/actions-setup-python@v4/
140[Gopls Update Checker/updateVersion]   🐳  docker exec cmd=[node /var/run/act/actions/actions-setup-python@v4/dist/setup/index.js] user= workdir=
141[Gopls Update Checker/updateVersion]   💬  ::debug::Python is expected to be installed into /opt/hostedtoolcache
142| [warning]Neither 'python-version' nor 'python-version-file' inputs were supplied. Attempting to find '.python-version' file.
143| [warning].python-version doesn't exist.
144[Gopls Update Checker/updateVersion]   🚧  ::warning::The `python-version` input is not set.  The version of Python currently in `PATH` will be used.
145[Gopls Update Checker/updateVersion]   ❓  ##[add-matcher]/run/act/actions/actions-setup-python@v4/.github/python.json
146[Gopls Update Checker/updateVersion]   ✅  Success - Main actions/setup-python@v4
147[Gopls Update Checker/updateVersion] ⭐ Run Main Update VERSION
148[Gopls Update Checker/updateVersion]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/4] user= workdir=
149[Gopls Update Checker/updateVersion]   ✅  Success - Main Update VERSION
150[Gopls Update Checker/updateVersion] ⭐ Run Main Update Schema and Settings
151[Gopls Update Checker/updateVersion]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/5] user= workdir=
152[Gopls Update Checker/updateVersion]   ✅  Success - Main Update Schema and Settings
153[Gopls Update Checker/updateVersion] ⭐ Run Post actions/setup-python@v4
154[Gopls Update Checker/updateVersion]   🐳  docker exec cmd=[node /var/run/act/actions/actions-setup-python@v4/dist/cache-save/index.js] user= workdir=
155[Gopls Update Checker/updateVersion]   ✅  Success - Post actions/setup-python@v4
156[Gopls Update Checker/updateVersion] ⭐ Run Post actions/setup-go@v3
157[Gopls Update Checker/updateVersion]   🐳  docker exec cmd=[node /var/run/act/actions/actions-setup-go@v3/dist/cache-save/index.js] user= workdir=
158| [command]/opt/hostedtoolcache/go/1.19.4/x64/bin/go env GOMODCACHE
159| [command]/opt/hostedtoolcache/go/1.19.4/x64/bin/go env GOCACHE
160| /root/go/pkg/mod
161| /root/.cache/go-build
162| Cache hit occurred on the primary key , not saving cache.
163[Gopls Update Checker/updateVersion]   ✅  Success - Post actions/setup-go@v3
164[Gopls Update Checker/updateVersion] 🏁  Job succeeded
```
</details>

### CODE REVIEW

File: .github/workflows/gopls_update_check.yaml [hunk 1]
```diff
@@ -10,38 +10,78 @@ jobs:
...
+      branch_name: ${{ steps.tagCompare.outputs.BRANCH_NAME }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
       - id: tagCompare
         name: Run script
+        shell: python {0}
         run: |
-          #!/bin/bash
+          import re
```
Comment (rchl) [inline .github/workflows/gopls_update_check.yaml:21]: wouldn't it be better to check it in as a normal script in the repo. I imagine it's quite painful to work on it from here (no syntax highlighting, type-checking, etc...).

File: .github/workflows/gopls_update_check.yaml [hunk 2]
```diff
@@ -52,12 +92,17 @@ jobs:
...
           fetch-depth: 0
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-go@v3
         with:
-          python-version: "3.10" # The version of Python to download (if necessary) and use.
+          go-version: "1.19"
+          check-latest: true
+          cache: true
+      - name: Install GOPLS
+        run: go install golang.org/x/tools/gopls@latest
```
Comment (rchl) [inline .github/workflows/gopls_update_check.yaml:101]: What is this step useful for? Is it to just check that it installs successfully? Shouldn't that be implied if it was released?

File: .github/workflows/gopls_update_check.yaml (hunk 1 above)
Comment (TerminalFi) [reply .github/workflows/gopls_update_check.yaml:21]: Sure. I'll do that.

File: .github/workflows/gopls_update_check.yaml (hunk 2 above)
Comment (TerminalFi) [reply .github/workflows/gopls_update_check.yaml:101]: Gopls is used in the script to generate the settings file.

File: scripts/compare_tags.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        v2_parts = [int(x) for x in v2.split(".")]
+
+        # Compare each part of the version
+        for i in range(min(len(v1_parts), len(v2_parts))):
+            if v1_parts[i] > v2_parts[i]:
+                return 0
+            elif v1_parts[i] < v2_parts[i]:
+                return 1
+        # If all parts are equal, return 1 if v2 has more parts
+        return 1 if len(v2_parts) > len(v1_parts) else 0
```
Comment (rchl) [inline scripts/compare_tags.py:55]: parsing into tuples and just comparing them generally works pretty well

**Review by rchl — APPROVED** [review]
haven't verified that any of this works

**Comment by TerminalFi** [comment]
> haven't verified that any of this works

I'll update local test output which is running  these locally and then switch to tuples and merge

**Pull request merged**